### PR TITLE
Create OSSF SECURITY-INSIGHTS.yml

### PR DESCRIPTION
**Pull request by mmorel-35**
Clomonitor requires a SECURITY-INSIGHTS.yml,
See https://clomonitor.io/projects/cncf/envoy#envoy_security

This is a first draft based https://github.com/ossf/security-insights-spec/blob/v1.0.0/specification.md and what I could find on the repository.

[CLOMonitor](https://clomonitor.io/projects/cncf/envoy)

Commit Message: Create OSSF SECURITY-INSIGHTS.yml
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax

File: SECURITY-INSIGHTS.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  schema-version: '1.0.0'
+  expiration-date: '2025-01-24T01:00:00.000Z'
+  last-updated: '2024-01-24'
+  last-reviewed: '2024-01-24'
+  project-url: https://github.com/envoyproxy/envoy
+  license: https://github.com/envoyproxy/envoy/blob/main/LICENSE
+project-lifecycle:
+  stage: active
+  bug-fixes-only: false
+  core-maintainers:
```
Comment (phlax) [inline SECURITY-INSIGHTS.yml:11]: this should include senior maintainers also

File: SECURITY-INSIGHTS.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  accepts-automated-pull-requests: true
+  code-of-conduct: https://github.com/envoyproxy/envoy/blob/main/CODE_OF_CONDUCT.md
+dependencies:
+  third-party-packages: true
+  dependencies-lists:
+    - https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps
+distribution-points:
+  - https://github.com/envoyproxy/envoy
+documentation:
+  - https://www.envoyproxy.io/
```
Comment (phlax) [inline SECURITY-INSIGHTS.yml:31]:
```suggestion
  - https://www.envoyproxy.io/docs
```

**Review by phlax — COMMENTED** [review]
very cool, thanks @mmorel-35 - couple of nits, and then lets add it

File: SECURITY-INSIGHTS.yml (hunk 1 above)
Comment (mmorel-35) [reply SECURITY-INSIGHTS.yml]: Done !

**Review by phlax — APPROVED** [review]
lgtm, thanks @mmorel-35

**Pull request merged**